statistics/handle/storage/save.go

```diff
@@ -77,6 +77,27 @@ func saveBucketsToStorage(ctx context.Context, exec sqlexec.SQLExecutor, sc *stm
 	if hg == nil {
 		return
 	}
+	selectForUpdate := new(strings.Builder)
+	selectForUpdate.WriteString("select mysql.stats_buckets where (table_id, is_index, hist_id, bucket_id) in (")
+	count := 0
+	for i := 0; i < len(hg.Buckets); i++ {
+		end := i + batchInsertSize
+		if end > len(hg.Buckets) {
+			end = len(hg.Buckets)
+		}
+		for j := i; j < end; j++ {
```

Comment:
The double loop is not required here because there is only one select-for-update statement.